File: src/main/java/no/nav/syfo/consumer/ws/XMLInntektsmelding.java
package no.nav.syfo.consumer.ws;

import lombok.Builder;
import lombok.Getter;
import lombok.Value;
import no.nav.syfo.domain.Periode;

import java.util.List;
import java.util.Optional;

@Value
@Builder
@Getter
public class XMLInntektsmelding {
    String arbeidsforholdId;
    List<Periode> perioder;
    String arbeidstakerFnr;
    Optional<String> virksomhetsnummer;
    Optional<String> arbeidsgiverPrivat;
    String aarsakTilInnsending;
}

File: src/main/kotlin/no/nav/syfo/consumer/ws/XMLInntektsmelding.kt
package no.nav.syfo.consumer.ws

import no.nav.syfo.domain.Periode


data class XMLInntektsmelding (
    val arbeidsforholdId: String? = null,
    val perioder: List<Periode>,
    val arbeidstakerFnr: String,
    val virksomhetsnummer: String? = null,
    val arbeidsgiverPrivat:String? = null,
    val aarsakTilInnsending: String
)
